### PR TITLE
FIX: Add the ability to accept `nil` values in NSPredicate+Extensions

### LINKED ISSUE #28: NSPredicate + dictionary should support `nil` values when forming predicates

### PR DESCRIPTION
**Pull request by kreeger**
Started using Flapjack in a side project and figured I'd add support for this while I was at it!

This closes #28. It's also _probably_ a breaking change, so while it's not necessary for us to immediately bump and create a new version, it should go out next as 0.7.

### CODE REVIEW

File: Flapjack/CoreData/Extensions/Dictionary+Extensions.swift [hunk 1]
```diff
@@ -8,13 +8,18 @@
 
 import Foundation
 
-internal extension Dictionary where Key == String, Value == Any {
+internal extension Dictionary where Key == String, Value == Optional<Any> {
```
Comment (mblackmon) [inline Flapjack/CoreData/Extensions/Dictionary+Extensions.swift:11]:
```suggestion
internal extension Dictionary where Key == String, Value == Any? {
```
Would you be open to changing this so the linter is happier?

**Review by mblackmon — CHANGES_REQUESTED** [review]
Looks good!  Just one suggestion above.

File: Flapjack/CoreData/Extensions/Dictionary+Extensions.swift (hunk 1 above)
Comment (kreeger) [reply Flapjack/CoreData/Extensions/Dictionary+Extensions.swift:11]: Oh for sure! Lemme take a gander

**Review by mblackmon — APPROVED** [review]

**Comment by mblackmon** [comment]
LGTM!

**Pull request merged**